mako/codegen.py

```diff
@@ -838,22 +838,36 @@ def visitControlLine(self, node):
                 text = node.text
             self.printer.writeline(text)
             children = node.get_children()
-            # this covers the three situations where we want to insert a pass:
-            #    1) a ternary control line with no children,
-            #    2) a primary control line with nothing but its own ternary
-            #          and end control lines, and
-            #    3) any control line with no content other than comments
-            if not children or (
-                all(
+
+            def should_add_pass(node, children):
+                # this covers the four situations
+                #   where we want to insert a pass:
+                # 1) a ternary control line with no children,
+                # 2) a primary control line with nothing but its own ternary
+                #       and end control lines, and
+                # 3) any control line with no content other than comments
+                # 4) the first control block with no content
+                #       other than comments
+                if not children:
+                    return True
+                elif all(
                     isinstance(c, (parsetree.Comment, parsetree.ControlLine))
                     for c in children
-                )
-                and all(
+                ) and all(
                     (node.is_ternary(c.keyword) or c.isend)
                     for c in children
                     if isinstance(c, parsetree.ControlLine)
-                )
-            ):
+                ):
+                    return True
+                else:
+                    for c in children:
+                        if isinstance(c, parsetree.Comment):
+                            continue
+                        elif isinstance(c, parsetree.ControlLine):
+                            return True
+                        return False
+
+            if should_add_pass(node, children):
```

Comment:
my comment didnt get transmitted from gerrit.



can we change this, so that we are using the same conditional from the old version, and just this last part with the "for c in children" is moved into a function def _search_for_control_line(). that way the form of the conditional stays the same and we move just that one check.

like this:

```py
if not children or (
    all(
        isinstance(c, (parsetree.Comment, parsetree.ControlLine))
        for c in children
    )
    and all(
        (node.is_ternary(c.keyword) or c.isend)
        for c in children
        if isinstance(c, parsetree.ControlLine)
    )
) or _search_for_control_line(children):
    self.printer.writeline("pass")

```